### DISCUSSION TITLE
Can multiple identical tunnels be opened between two ipsec servers?

### DISCUSSION THREAD

**Discussion by jiawen94** [strongswan/strongswan#2014] [Q&A]
In order to improve the throughput performance of ipsec tunnels (because the cpu load is too low), I plan to deploy multiple ipsec tunnels between two servers. After I start the ipsec tunnel, I found that the server has only one tunnel active state. Is it possible that ipsec is designed to have only one tunnel between two servers?

**Comment by tobiasbrunner** [comment]
You might want to have a look at the _per-cpu-sas-poc_ branch (and [the draft](https://datatracker.ietf.org/doc/html/draft-ietf-ipsecme-multi-sa-performance) on which it's based, as that also references the Linux patches, and maybe [this email](https://linux-ipsec.org/pipermail/devel/2023/000355.html)).

**Reply by jiawen94** [reply]
I'll try the per-cpu-sas-poc branch. Great point about utilizing the CPU's cores. Any plans to merge into the main branch?
By the way, Is strongswan designed in such a way that multiple tunnels (same algorithms, protocols, gateways, and subnets) are not allowed between two strongwan servers?

**Reply by tobiasbrunner** [reply]
> Any plans to merge into the main branch?

At some point in the future, sure.





> By the way, Is strongswan designed in such a way that multiple tunnels (same algorithms, protocols, gateways, and subnets) are not allowed between two strongwan servers?

Yes, duplicate CHILD_SAs are prevented as initiator since 5.9.6 as they are useless and were often created accidentally. To create "duplicates" you have to make them unique e.g. by using marks or interface IDs (or labels, or CPU IDs like the branch does), which then also requires marking/routing traffic accordingly.
